site/faq.md

```diff
@@ -247,11 +246,12 @@ happen:
    if you've only just started using a particular image in a workload.
  - Flux can't get suitable credentials for the image repository. At
    present, it looks at `imagePullSecret`s attached to workloads,
-   service accounts and a Docker config file if you mount one into the fluxd container
-   (see the [command-line usage](./daemon.md)).
- - Flux doesn't know how to obtain registry credentials for ECR. A
-   workaround is described in
-   [weaveworks/flux#539](https://github.com/weaveworks/flux/issues/539#issuecomment-394588423)
+   service accounts, platform-provided credentials on GCP or AWS, and
+   a Docker config file if you mount one into the fluxd container (see
+   the [command-line usage](./daemon.md)).
+ - When using images in ECR, from AWS, the IAM account used to run the
```

Comment:
I'd rephrase as:
```
 - When using images in AWS ECR, the EC2 instance role of the node
  fluxd container runs on must have permissions to query the ECR registry
  or registries in question. In turns, this is a requirement for kubelet also,
  so if your cluster can run containers from ECR, Flux should work too.
```

Reply:
Is it definitely the EC2 instance role that decides whether you have access? I confess I skipped lightly over the question of how the container gets any IAM permissions at all (and Just Made It Work TM). I'm happy to be more specific about the mechanism.